src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,640 @@
+/*---------------------------------------------------------------
+ * Programmer(s): Mustafa Aggul @ SMU
+ *---------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *---------------------------------------------------------------
+ * This is the implementation file for the optional input and
+ * output functions for the ARKODE LSRKStep time stepper module.
+ *--------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <sunadaptcontroller/sunadaptcontroller_soderlind.h>
+#include "arkode_lsrkstep_impl.h"
+
+/*===============================================================
+  Exported optional input functions.
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  LSRKStepSetMethod sets method
+    ARKODE_LSRK_RKC_2
+    ARKODE_LSRK_RKL_2
+    ARKODE_LSRK_RKG_2
+    ARKODE_LSRK_SSP_S_2
+    ARKODE_LSRK_SSP_S_3
+    ARKODE_LSRK_SSP_10_4
+  ---------------------------------------------------------------*/
+int LSRKStepSetMethod(void* arkode_mem, ARKODE_LSRKMethodType method)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (method)
+  {
+  case ARKODE_LSRK_RKC_2:
+    ark_mem->step          = lsrkStep_TakeStepRKC;
+    step_mem->LSRKmethod   = ARKODE_LSRK_RKC_2;
+    step_mem->nfusedopvecs = 5;
+    step_mem->q = ark_mem->hadapt_mem->q = 2;
+    step_mem->p = ark_mem->hadapt_mem->p = 2;
+    break;
+  case ARKODE_LSRK_RKL_2:
+    ark_mem->step          = lsrkStep_TakeStepRKL;
+    step_mem->LSRKmethod   = ARKODE_LSRK_RKL_2;
+    step_mem->nfusedopvecs = 5;
+    step_mem->q = ark_mem->hadapt_mem->q = 2;
+    step_mem->p = ark_mem->hadapt_mem->p = 2;
+    break;
+  // case ARKODE_LSRK_RKG_2:
```

Comment:
Are we leaving this code here because we may come back to it later? Generally, we remove commented out code.  If we want to leave it in for further development (in the near future), then please add a `TODO(YOUR INITIALS): <description>` comment.